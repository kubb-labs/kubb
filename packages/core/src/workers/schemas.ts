import { z } from 'zod'

import type { ZodStringDef, ZodType } from 'zod'
import type { KubbFile } from '../FileManager.ts'

export const importSchema = z.object({
  name: z.union([z.string(), z.array(z.string())]),
  path: z.string(),
  isTypeOnly: z.boolean().optional(),
  isNameSpace: z.boolean().optional(),
  root: z.string().optional(),
})

export const exportSchema = z.object({
  name: z.union([z.string(), z.array(z.string())]).optional(),
  path: z.string(),
  isTypeOnly: z.boolean().optional(),
  asAlias: z.boolean().optional(),
})

export const fileSchema = z.object({
  id: z.string().optional(),
  baseName: z.string() as ZodType<KubbFile.BaseName, ZodStringDef>,
  path: z.string(),
  source: z.string(),
  imports: z.array(importSchema).optional(),
  exports: z.array(exportSchema).optional(),
  override: z.boolean().optional(),
  meta: z.object({}).optional(),
  env: z.object({}).optional(),
})
