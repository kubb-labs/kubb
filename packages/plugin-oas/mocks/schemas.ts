import { type Schema, schemaKeywords } from '../src/SchemaMapper'

const basic: Array<{ name: string; schema: Schema }> = [
  {
    name: 'any',
    schema: {
      keyword: schemaKeywords.any,
    },
  },
  {
    name: 'unknown',
    schema: {
      keyword: schemaKeywords.unknown,
    },
  },
  {
    name: 'string',
    schema: {
      keyword: schemaKeywords.string,
    },
  },
  {
    name: 'number',
    schema: {
      keyword: schemaKeywords.number,
    },
  },
  {
    name: 'integer',
    schema: {
      keyword: schemaKeywords.integer,
    },
  },
  {
    name: 'boolean',
    schema: {
      keyword: schemaKeywords.boolean,
    },
  },
  {
    name: 'primitiveDate',
    schema: {
      keyword: schemaKeywords.date,
      args: {
        type: 'date',
      },
    },
  },
  {
    name: 'date',
    schema: {
      keyword: schemaKeywords.date,
      args: {
        type: 'string',
      },
    },
  },
  {
    name: 'time',
    schema: {
      keyword: schemaKeywords.time,
      args: {
        type: 'string',
      },
    },
  },
  {
    name: 'stringOffset',
    schema: {
      keyword: schemaKeywords.datetime,
      args: {
        offset: true,
      },
    },
  },
  {
    name: 'stringLocal',
    schema: {
      keyword: schemaKeywords.datetime,
      args: {
        local: true,
      },
    },
  },
  {
    name: 'datetime',
    schema: {
      keyword: schemaKeywords.datetime,
      args: {
        offset: false,
      },
    },
  },
  {
    name: 'nullable',
    schema: {
      keyword: schemaKeywords.nullable,
    },
  },
  {
    name: 'undefined',
    schema: {
      keyword: schemaKeywords.undefined,
    },
  },
  {
    name: 'min',
    schema: {
      keyword: schemaKeywords.min,
      args: 2,
    },
  },
  {
    name: 'max',
    schema: {
      keyword: schemaKeywords.max,
      args: 2,
    },
  },
  {
    name: 'matchesReg',
    schema: {
      keyword: schemaKeywords.matches,
      args: '/node_modules/', // pure regexp
    },
  },
  {
    name: 'matches',
    schema: {
      keyword: schemaKeywords.matches,
      args: '^[A-Z]{2}$',
    },
  },
  {
    name: 'ref',
    schema: {
      keyword: schemaKeywords.ref,
      args: {
        name: 'Pet',
        path: './pet.ts',
      },
    },
  },
  {
    name: 'enum',
    schema: {
      keyword: schemaKeywords.enum,
      args: {
        name: 'enum',
        typeName: 'Enum',
        asConst: false,
        items: [
          { name: 'A', value: 'A', format: schemaKeywords.string },
          { name: 'B', value: 'B', format: schemaKeywords.string },
          { name: 'C', value: 'C', format: schemaKeywords.string },
          { name: 2, value: 2, format: schemaKeywords.number },
        ],
      },
    },
  },

  {
    name: 'tuple',
    schema: {
      keyword: schemaKeywords.tuple,
      args: {
        items: [],
      },
    },
  },
  {
    name: 'tupleMulti',
    schema: {
      keyword: schemaKeywords.tuple,
      args: {
        items: [{ keyword: schemaKeywords.string }, { keyword: schemaKeywords.number }],
      },
    },
  },
  {
    name: 'array',
    schema: {
      keyword: schemaKeywords.array,
      args: {
        items: [
          {
            keyword: schemaKeywords.union,
            args: [{ keyword: schemaKeywords.number }, { keyword: schemaKeywords.string }],
          },
        ],
      },
    },
  },
  {
    name: 'arrayEmpty',
    schema: {
      keyword: schemaKeywords.array,
      args: {
        items: [],
      },
    },
  },
  {
    name: 'arrayRef',
    schema: {
      keyword: schemaKeywords.array,
      args: {
        items: [
          {
            keyword: schemaKeywords.ref,
            args: { name: 'Pet', path: './pet.ts' },
          },
        ],
      },
    },
  },
  {
    name: 'arrayAdvanced',
    schema: {
      keyword: schemaKeywords.array,
      args: {
        items: [{ keyword: schemaKeywords.min, args: 1 }, { keyword: schemaKeywords.max, args: 10 }, { keyword: schemaKeywords.number }],
        min: 3,
        max: 10,
      },
    },
  },
  {
    name: 'arrayRegex',
    schema: {
      keyword: schemaKeywords.array,
      args: {
        items: [{ keyword: schemaKeywords.matches, args: '^[a-zA-Z0-9]{1,13}$' }, { keyword: schemaKeywords.string }],
        min: 3,
        max: 10,
      },
    },
  },
  {
    name: 'union',
    schema: {
      keyword: schemaKeywords.union,
      args: [{ keyword: schemaKeywords.string }, { keyword: schemaKeywords.number }],
    },
  },
  {
    name: 'unionOne',
    schema: {
      keyword: schemaKeywords.union,
      args: [{ keyword: schemaKeywords.string }],
    },
  },
  {
    name: 'catchall',
    schema: {
      keyword: schemaKeywords.object,
      args: {
        properties: {},
        additionalProperties: [
          {
            keyword: schemaKeywords.ref,
            args: { name: 'Pet', path: './Pet.ts' },
          },
        ],
      },
    },
  },
  {
    name: 'and',
    schema: {
      keyword: schemaKeywords.and,
      args: [{ keyword: schemaKeywords.string }, { keyword: schemaKeywords.number }],
    },
  },
  {
    name: 'object',
    schema: {
      keyword: schemaKeywords.object,
      args: {
        properties: {
          firstName: [{ keyword: schemaKeywords.string }, { keyword: schemaKeywords.min, args: 2 }],
          address: [{ keyword: schemaKeywords.string }, { keyword: schemaKeywords.nullable }, { keyword: schemaKeywords.describe, args: '"Your address"' }],
        },
        additionalProperties: [],
      },
    },
  },
  {
    name: 'objectOptional',
    schema: {
      keyword: schemaKeywords.object,
      args: {
        properties: {
          firstName: [{ keyword: schemaKeywords.string }, { keyword: schemaKeywords.optional }, { keyword: schemaKeywords.min, args: 2 }],
          address: [{ keyword: schemaKeywords.string }, { keyword: schemaKeywords.nullable }, { keyword: schemaKeywords.describe, args: '"Your address"' }],
        },
        additionalProperties: [],
      },
    },
  },
  {
    name: 'objectArray',
    schema: {
      keyword: schemaKeywords.object,
      args: {
        properties: {
          ids: [
            {
              keyword: schemaKeywords.array,
              args: {
                items: [{ keyword: schemaKeywords.matches, args: '^[a-zA-Z0-9]{1,13}$' }, { keyword: schemaKeywords.string }],
                min: 3,
                max: 10,
              },
            },
          ],
        },
        additionalProperties: [],
      },
    },
  },
  {
    name: 'objectDates',
    schema: {
      keyword: schemaKeywords.object,
      args: {
        properties: {
          dateTime: [{ keyword: schemaKeywords.datetime, args: { offset: true } }],
          date: [{ keyword: schemaKeywords.date, args: { type: 'string' } }],
          time: [{ keyword: schemaKeywords.time, args: { type: 'string' } }],
          nativeDate: [{ keyword: schemaKeywords.date, args: { type: 'date' } }],
        },
        additionalProperties: [],
      },
    },
  },
  {
    name: 'objectAnd',
    schema: {
      keyword: schemaKeywords.object,
      args: {
        properties: {
          firstName: [
            { keyword: schemaKeywords.deprecated },
            { keyword: schemaKeywords.default, args: 'test' },
            {
              keyword: schemaKeywords.min,
              args: 2,
            },
            {
              keyword: schemaKeywords.string,
            },
          ],
          age: [
            { keyword: schemaKeywords.example, args: '2' },
            { keyword: schemaKeywords.default, args: 2 },
            {
              keyword: schemaKeywords.min,
              args: 3,
            },
            {
              keyword: schemaKeywords.number,
            },
          ],
          address: [
            {
              keyword: schemaKeywords.and,
              args: [{ keyword: schemaKeywords.string }, { keyword: schemaKeywords.number }],
            },
            { keyword: schemaKeywords.nullable },
            { keyword: schemaKeywords.describe, args: 'Your address' },
          ],
        },
        additionalProperties: [],
      },
    },
  },
  {
    name: 'objectEnum',
    schema: {
      keyword: schemaKeywords.object,
      args: {
        properties: {
          version: [
            {
              keyword: schemaKeywords.schema,
              args: {
                format: 'string',
                type: 'string',
              },
            },
            {
              keyword: schemaKeywords.enum,
              args: {
                name: 'enum',
                typeName: 'Enum',
                asConst: false,
                items: [
                  { name: 'A', value: 'A', format: schemaKeywords.string },
                  { name: 'B', value: 'B', format: schemaKeywords.string },
                  { name: 'C', value: 'C', format: schemaKeywords.string },
                  { name: 2, value: 2, format: schemaKeywords.number },
                ],
              },
            },
            {
              keyword: schemaKeywords.min,
              args: 4,
            },
            { keyword: schemaKeywords.describe, args: 'Your address' },
          ],
        },
        additionalProperties: [],
      },
    },
  },
  {
    name: 'objectEmpty',
    schema: {
      keyword: schemaKeywords.object,
      args: {
        properties: {},
        additionalProperties: [],
      },
    },
  },
  {
    name: 'default',
    schema: {
      keyword: schemaKeywords.default,
    },
  },
  {
    name: 'default',
    schema: {
      keyword: schemaKeywords.default,
      args: 'default',
    },
  },
  {
    name: 'blob',
    schema: {
      keyword: schemaKeywords.blob,
    },
  },
]

const full: Array<{ name: string; schema: Schema[] }> = [
  {
    name: 'Upload',
    schema: [
      {
        keyword: schemaKeywords.blob,
      },
    ],
  },
  {
    name: 'PageSizeNumber',
    schema: [
      {
        keyword: schemaKeywords.number,
      },
      {
        keyword: schemaKeywords.default,
        args: 10,
      },
    ],
  },
  {
    name: 'PageSizeInteger',
    schema: [
      {
        keyword: schemaKeywords.integer,
      },
      {
        keyword: schemaKeywords.default,
        args: 10,
      },
    ],
  },
  {
    name: 'Object',
    schema: [
      { keyword: schemaKeywords.nullable },
      { keyword: schemaKeywords.describe, args: 'Your address' },
      {
        keyword: schemaKeywords.object,
        args: {
          properties: {
            firstName: [
              { keyword: schemaKeywords.deprecated },
              { keyword: schemaKeywords.default, args: 'test' },
              {
                keyword: schemaKeywords.min,
                args: 2,
              },
              {
                keyword: schemaKeywords.string,
              },
            ],
            age: [
              { keyword: schemaKeywords.example, args: '2' },
              { keyword: schemaKeywords.default, args: 2 },
              {
                keyword: schemaKeywords.min,
                args: 2,
              },
              {
                keyword: schemaKeywords.number,
              },
            ],
            address: [
              {
                keyword: schemaKeywords.and,
                args: [{ keyword: schemaKeywords.string }, { keyword: schemaKeywords.number }],
              },
              { keyword: schemaKeywords.nullable },
              { keyword: schemaKeywords.describe, args: 'Your address' },
            ],
          },
          additionalProperties: [],
        },
      },
    ],
  },
  {
    name: 'Order',
    schema: [
      {
        keyword: schemaKeywords.schema,
        args: {
          type: 'object',
        },
      },
      {
        keyword: schemaKeywords.object,
        args: {
          properties: {
            status: [
              {
                keyword: schemaKeywords.enum,
                args: {
                  name: 'orderStatus',
                  asConst: false,
                  typeName: 'OrderStatus',
                  items: [
                    { name: 'Placed', value: 'placed', format: 'string' },
                    { name: 'Approved', value: 'approved', format: 'string' },
                  ],
                },
              },
            ],
          },
          additionalProperties: [],
        },
      },
    ],
  },
  {
    name: 'Record',
    schema: [
      {
        keyword: schemaKeywords.object,
        args: {
          properties: {},
          additionalProperties: [
            {
              keyword: schemaKeywords.integer,
            },
            {
              keyword: schemaKeywords.schema,
              args: {
                type: 'integer',
                format: 'int32',
              },
            },
            {
              keyword: schemaKeywords.optional,
            },
          ],
        },
      },
      {
        keyword: schemaKeywords.schema,
        args: {
          type: 'integer',
          format: 'int32',
        },
      },
      {
        keyword: schemaKeywords.optional,
      },
    ],
  },
]

export const schemas = {
  basic,
  full,
}
