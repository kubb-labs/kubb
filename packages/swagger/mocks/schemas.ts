import { schemaKeywords, type Schema } from '../src/SchemaMapper'

const basic: Array<{ name: string; schema: Schema }> = [
  {
    name: 'any',
    schema: {
      keyword: schemaKeywords.any,
    },
  },
  {
    name: 'unknown',
    schema: {
      keyword: schemaKeywords.unknown,
    },
  },
  {
    name: schemaKeywords.string,
    schema: {
      keyword: schemaKeywords.string,
    },
  },
  {
    name: schemaKeywords.number,
    schema: {
      keyword: schemaKeywords.number,
    },
  },
  {
    name: 'boolean',
    schema: {
      keyword: schemaKeywords.boolean,
    },
  },
  {
    name: 'date',
    schema: {
      keyword: schemaKeywords.date,
    },
  },
  {
    name: 'datetime',
    schema: {
      keyword: schemaKeywords.datetime,
    },
  },
  {
    name: 'nullable',
    schema: {
      keyword: schemaKeywords.nullable,
    },
  },
  {
    name: 'undefined',
    schema: {
      keyword: schemaKeywords.undefined,
    },
  },
  {
    name: schemaKeywords.min,
    schema: {
      keyword: schemaKeywords.min,
      args: 2,
    },
  },
  {
    name: 'max',
    schema: {
      keyword: schemaKeywords.max,
      args: 2,
    },
  },
  {
    name: 'matchesReg',
    schema: {
      keyword: schemaKeywords.matches,
      args: '/node_modules/', // pure regexp
    },
  },
  {
    name: 'matches',
    schema: {
      keyword: schemaKeywords.matches,
      args: '^[A-Z]{2}$',
    },
  },
  {
    name: 'ref',
    schema: {
      keyword: schemaKeywords.ref,
      args: {
        name: 'Pet',
        path: './pet.ts',
      },
    },
  },
  {
    name: 'enum',
    schema: {
      keyword: schemaKeywords.enum,
      args: {
        name: 'enum',
        typeName: 'Enum',
        asConst: false,
        items: [
          { name: 'A', value: 'A', format: schemaKeywords.string },
          { name: 'B', value: 'B', format: schemaKeywords.string },
          { name: 'C', value: 'C', format: schemaKeywords.string },
          { name: 2, value: 2, format: schemaKeywords.number },
        ],
      },
    },
  },
  {
    name: 'tuple',
    schema: {
      keyword: schemaKeywords.tuple,
      args: [],
    },
  },
  {
    name: 'tupleMulti',
    schema: {
      keyword: schemaKeywords.tuple,
      args: [{ keyword: schemaKeywords.string }, { keyword: schemaKeywords.number }],
    },
  },
  {
    name: 'array',
    schema: {
      keyword: schemaKeywords.array,
      args: {
        items: [
          {
            keyword: schemaKeywords.union,
            args: [{ keyword: schemaKeywords.number }, { keyword: schemaKeywords.string }],
          },
        ],
      },
    },
  },
  {
    name: 'arrayEmpty',
    schema: {
      keyword: schemaKeywords.array,
      args: {
        items: [],
      },
    },
  },
  {
    name: 'arrayRef',
    schema: {
      keyword: schemaKeywords.array,
      args: {
        items: [
          {
            keyword: schemaKeywords.ref,
            args: { name: 'Pet', path: './pet.ts' },
          },
        ],
      },
    },
  },
  {
    name: 'arrayAdvanced',
    schema: {
      keyword: schemaKeywords.array,
      args: {
        items: [{ keyword: schemaKeywords.min, args: 1 }, { keyword: schemaKeywords.number }],
        min: 3,
        max: 10,
      },
    },
  },
  {
    name: 'union',
    schema: {
      keyword: schemaKeywords.union,
      args: [{ keyword: schemaKeywords.string }, { keyword: schemaKeywords.number }],
    },
  },
  {
    name: 'unionOne',
    schema: {
      keyword: schemaKeywords.union,
      args: [{ keyword: schemaKeywords.string }],
    },
  },
  {
    name: 'catchall',
    schema: {
      keyword: schemaKeywords.object,
      args: {
        properties: {},
        additionalProperties: [
          {
            keyword: schemaKeywords.ref,
            args: { name: 'Pet', path: './Pet.ts' },
          },
        ],
      },
    },
  },
  {
    name: 'and',
    schema: {
      keyword: schemaKeywords.and,
      args: [{ keyword: schemaKeywords.string }, { keyword: schemaKeywords.number }],
    },
  },
  {
    name: 'object',
    schema: {
      keyword: schemaKeywords.object,
      args: {
        properties: {
          firstName: [{ keyword: schemaKeywords.string }, { keyword: schemaKeywords.min, args: 2 }],
          address: [{ keyword: schemaKeywords.string }, { keyword: schemaKeywords.nullable }, { keyword: schemaKeywords.describe, args: '"Your address"' }],
        },
        additionalProperties: [],
      },
    },
  },
  {
    name: 'objectOptional',
    schema: {
      keyword: schemaKeywords.object,
      args: {
        properties: {
          firstName: [{ keyword: schemaKeywords.string }, { keyword: schemaKeywords.optional }, { keyword: schemaKeywords.min, args: 2 }],
          address: [{ keyword: schemaKeywords.string }, { keyword: schemaKeywords.nullable }, { keyword: schemaKeywords.describe, args: '"Your address"' }],
        },
        additionalProperties: [],
      },
    },
  },
  {
    name: 'objectAnd',
    schema: {
      keyword: schemaKeywords.object,
      args: {
        properties: {
          firstName: [
            { keyword: schemaKeywords.deprecated },
            { keyword: schemaKeywords.default, args: 'test' },
            {
              keyword: schemaKeywords.min,
              args: 2,
            },
            {
              keyword: schemaKeywords.string,
            },
          ],
          age: [
            { keyword: schemaKeywords.example, args: '2' },
            { keyword: schemaKeywords.default, args: 2 },
            {
              keyword: schemaKeywords.min,
              args: 2,
            },
            {
              keyword: schemaKeywords.number,
            },
          ],
          address: [
            {
              keyword: schemaKeywords.and,
              args: [{ keyword: schemaKeywords.string }, { keyword: schemaKeywords.number }],
            },
            { keyword: schemaKeywords.nullable },
            { keyword: schemaKeywords.describe, args: 'Your address' },
          ],
        },
        additionalProperties: [],
      },
    },
  },
  {
    name: 'objectEmpty',
    schema: {
      keyword: schemaKeywords.object,
      args: {
        properties: {},
        additionalProperties: [],
      },
    },
  },
  {
    name: 'default',
    schema: {
      keyword: schemaKeywords.default,
    },
  },
  {
    name: 'default',
    schema: {
      keyword: schemaKeywords.default,
      args: 'default',
    },
  },
  {
    name: 'blob',
    schema: {
      keyword: schemaKeywords.blob,
    },
  },
]

const full: Array<{ name: string; schema: Schema[] }> = [
  {
    name: 'Upload',
    schema: [
      {
        keyword: schemaKeywords.blob,
      },
    ],
  },
  {
    name: 'Object',
    schema: [
      { keyword: schemaKeywords.nullable },
      { keyword: schemaKeywords.describe, args: 'Your address' },
      {
        keyword: schemaKeywords.object,
        args: {
          properties: {
            firstName: [
              { keyword: schemaKeywords.deprecated },
              { keyword: schemaKeywords.default, args: 'test' },
              {
                keyword: schemaKeywords.min,
                args: 2,
              },
              {
                keyword: schemaKeywords.string,
              },
            ],
            age: [
              { keyword: schemaKeywords.example, args: '2' },
              { keyword: schemaKeywords.default, args: 2 },
              {
                keyword: schemaKeywords.min,
                args: 2,
              },
              {
                keyword: schemaKeywords.number,
              },
            ],
            address: [
              {
                keyword: schemaKeywords.and,
                args: [{ keyword: schemaKeywords.string }, { keyword: schemaKeywords.number }],
              },
              { keyword: schemaKeywords.nullable },
              { keyword: schemaKeywords.describe, args: 'Your address' },
            ],
          },
          additionalProperties: [],
        },
      },
    ],
  },
  {
    name: 'Order',
    schema: [
      {
        keyword: schemaKeywords.type,
        args: 'object',
      },
      {
        keyword: schemaKeywords.object,
        args: {
          properties: {
            status: [
              {
                keyword: schemaKeywords.enum,
                args: {
                  name: 'orderStatus',
                  asConst: false,
                  typeName: 'OrderStatus',
                  items: [
                    { name: 'Placed', value: 'placed', format: 'string' },
                    { name: 'Approved', value: 'approved', format: 'string' },
                  ],
                },
              },
            ],
          },
          additionalProperties: [],
        },
      },
    ],
  },
  {
    name: 'Record',
    schema: [
      {
        keyword: schemaKeywords.object,
        args: {
          properties: {},
          additionalProperties: [
            {
              keyword: schemaKeywords.integer,
            },
            {
              keyword: schemaKeywords.type,
              args: 'integer',
            },
            {
              keyword: schemaKeywords.format,
              args: 'int32',
            },
            {
              keyword: schemaKeywords.optional,
            },
          ],
        },
      },
      {
        keyword: 'type',
        args: 'object',
      },
      {
        keyword: 'optional',
      },
    ],
  },
]

export const schemas = {
  basic,
  full,
}
