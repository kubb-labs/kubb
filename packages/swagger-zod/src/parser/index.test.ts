import { schemas } from '../../../plugin-oas/mocks/schemas.ts'

import * as parserZod from './index.ts'

describe('zod parse', () => {
  test.each(schemas.basic)('$name', ({ name, schema }) => {
    const text = parserZod.parse(undefined, schema, { name })
    expect(text).toMatchSnapshot()
  })
})
