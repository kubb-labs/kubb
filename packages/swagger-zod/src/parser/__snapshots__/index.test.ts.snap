// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`zod parse > 'and' 1`] = `"z.string().and(z.number())"`;

exports[`zod parse > 'any' 1`] = `"z.any()"`;

exports[`zod parse > 'array' 1`] = `"z.array(z.union([z.string(), z.number()]))"`;

exports[`zod parse > 'arrayAdvanced' 1`] = `"z.array(z.number().min(1).max(10)).min(3).max(10)"`;

exports[`zod parse > 'arrayEmpty' 1`] = `"z.array()"`;

exports[`zod parse > 'arrayRef' 1`] = `"z.array(z.lazy(() => Pet))"`;

exports[`zod parse > 'arrayRegex' 1`] = `"z.array(z.string().regex(new RegExp('^[a-zA-Z0-9]{1,13}$'))).min(3).max(10)"`;

exports[`zod parse > 'blob' 1`] = `"z.string()"`;

exports[`zod parse > 'boolean' 1`] = `"z.boolean()"`;

exports[`zod parse > 'catchall' 1`] = `"z.object({}).catchall(z.lazy(() => Pet))"`;

exports[`zod parse > 'date' 1`] = `"z.string().date()"`;

exports[`zod parse > 'datetime' 1`] = `"z.string().datetime()"`;

exports[`zod parse > 'default' 1`] = `".default()"`;

exports[`zod parse > 'default' 2`] = `".default(default)"`;

exports[`zod parse > 'enum' 1`] = `"z.enum(["A", "B", "C", "2"])"`;

exports[`zod parse > 'integer' 1`] = `"z.number().int()"`;

exports[`zod parse > 'matches' 1`] = `".regex(new RegExp('^[A-Z]{2}$'))"`;

exports[`zod parse > 'matchesReg' 1`] = `".regex(new RegExp('node_modules'))"`;

exports[`zod parse > 'max' 1`] = `".max(2)"`;

exports[`zod parse > 'min' 1`] = `".min(2)"`;

exports[`zod parse > 'nullable' 1`] = `".nullable()"`;

exports[`zod parse > 'number' 1`] = `"z.number()"`;

exports[`zod parse > 'object' 1`] = `"z.object({"firstName": z.string().min(2),"address": z.string().describe("Your address").nullable()})"`;

exports[`zod parse > 'objectAnd' 1`] = `"z.object({"firstName": z.string().min(2).default(test),"age": z.number().min(3).default(2),"address": z.string().and(z.number()).describe("Your address").nullable()})"`;

exports[`zod parse > 'objectArray' 1`] = `"z.object({"ids": z.array(z.string().regex(new RegExp('^[a-zA-Z0-9]{1,13}$'))).min(3).max(10)})"`;

exports[`zod parse > 'objectDates' 1`] = `"z.object({"dateTime": z.string().datetime({ offset: true }),"date": z.string().date(),"time": z.string().time(),"nativeDate": z.date()})"`;

exports[`zod parse > 'objectEmpty' 1`] = `"z.object({})"`;

exports[`zod parse > 'objectEnum' 1`] = `"z.object({"version": z.enum(["A", "B", "C", "2"]).min(4).describe("Your address")})"`;

exports[`zod parse > 'objectOptional' 1`] = `"z.object({"firstName": z.string().min(2).optional(),"address": z.string().describe("Your address").nullable()})"`;

exports[`zod parse > 'primitiveDate' 1`] = `"z.date()"`;

exports[`zod parse > 'ref' 1`] = `"z.lazy(() => Pet)"`;

exports[`zod parse > 'string' 1`] = `"z.string()"`;

exports[`zod parse > 'stringLocal' 1`] = `"z.string().datetime({ local: true })"`;

exports[`zod parse > 'stringOffset' 1`] = `"z.string().datetime({ offset: true })"`;

exports[`zod parse > 'time' 1`] = `"z.string().time()"`;

exports[`zod parse > 'tuple' 1`] = `"z.tuple([])"`;

exports[`zod parse > 'tupleMulti' 1`] = `"z.tuple([z.string(), z.number()])"`;

exports[`zod parse > 'undefined' 1`] = `"z.undefined()"`;

exports[`zod parse > 'union' 1`] = `"z.union([z.string(), z.number()])"`;

exports[`zod parse > 'unionOne' 1`] = `"z.string()"`;

exports[`zod parse > 'unknown' 1`] = `"z.unknown()"`;
